src/main/java/uk/gov/ons/registers/cdap/service/JSONHelper.java

```diff
@@ -0,0 +1,54 @@
+package uk.gov.ons.registers.cdap.service;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import uk.gov.ons.registers.cdap.service.tablecolumns.CompanyHouseTable;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+
+class JSONHelper {
+    private JSONHelper(){
+        throw new AssertionError("This class is not meant to be instantiated");
+    }
+
+    /**
+     * Method that takes byte[] HashMap of results and returns a decoded JSON object with the results
+     * Following the JSON structure of Admin Data Service
+     *
+     * String referenceColumnName is used to set the ID of the JSON object from the rowkey column in dataset being used
+     */
+    static JsonElement byteMapToGenericJSON(Map<byte[], byte[]> hashMap, String referenceColumnName){
+
+        Gson gson = new Gson();
+        JsonObject jsonData = new JsonObject();
+        JsonObject jsonVariables = new JsonObject();
+
+        // Iterates thought results and casts the byte[] objects to Strings to a JSON Object
+        for (Map.Entry<byte[], byte[]> entry : hashMap.entrySet()) {
+            String keyString;
+            String valueString;
+
+            keyString = new String(entry.getKey(), StandardCharsets.UTF_8);
+            valueString = new String(entry.getValue(), StandardCharsets.UTF_8);
+
+            //Adds id and period based on JSON structure
+            if (keyString.equals(CompanyHouseTable.PERIOD_COLUMN)) {
+                jsonData.add(CompanyHouseTable.PERIOD_COLUMN, gson.toJsonTree(valueString));
+            }
+            if (keyString.equals(referenceColumnName)) {
+                jsonData.add(CompanyHouseTable.ID_COLUMN, gson.toJsonTree(valueString));
+            }
+
+            //All other values to separate JSON object
+            jsonVariables.add(keyString, gson.toJsonTree(valueString));
```

Comment:
This would be a great place to add readability by using a switch case for the two if statements as well as the default case.

So something like:
```java
            switch(keyString){
                case CompanyHouseTable.PERIOD_COLUMN:
                    jsonData.add(CompanyHouseTable.PERIOD_COLUMN, gson.toJsonTree(valueString));
					break;
				case ....:
						.....
				default:
					......	
            }
```

Besides readability, javac also generates more efficient bytecode for a switch-case on String compared to bunch of if-then-else (Admittedly, in this code we only have a bunch of ifs, so I am not entirely sure if this benefit would apply. However, readability itself is a good enough reason IMHO).

Reply:
Looking at this further although a case would be more readable, I can't see a way of using it because One of the columns is set as an argument that passed into the method, I do this as the generic ID column is based of different columns for example the company house ID is companynumber where as the VAT ID is the vatref column.

Reply:
Hmm, in that case, perhaps we want to refactor a bit more. I am not a fan of the fact that JSONHelper, a generic object, is tightly coupled with the Service(s). It would be better to delegate the logic of deciding the mapping between `keyString` and the columnName to the Service classes themselves.

One way of doing this, would be to let the client of this class, pass in the logic which this class then simply calls.

For example, in CompanyHouseService you would define this:
```java
    static Function<String, String> mapSourceColumnName = (sourceColumnName) -> {
        String destinationColumnName;

        switch (sourceColumnName) {
            case CompanyHouseTable.COMPANY_NUMBER_COLUMN:
                destinationColumnName = CompanyHouseTable.ID_COLUMN;
                break;
            default:
                destinationColumnName = sourceColumnName;
        }

        return destinationColumnName;
    };
```

The second param to the function definition would accordingly change to `Function<String, String> mapColumnName)` and the above code would be simply replaced from what it is now to simply:

```java
jsonVariables.add(mapColumnName.apply(keyString),gson.toJsonTree(valueString));
```